src/gui/pedalboard/cables/Cable.cpp

```diff
@@ -35,6 +35,18 @@ bool Cable::hitTest (int x, int y)
     return false;
 }
 
+ConnectionInfo* Cable::getConnectionInfo()
```

Comment:
I did some thinking about passing a ConnectionInfo object into the ProcessorStore::createProcFromCableClickList () function, like you suggested [here](https://github.com/Chowdhury-DSP/BYOD/pull/182#discussion_r945149583). I thought it might be easier to read if a pointer was just passed right from the start? I could be totally off on this so let me know if there are better reasons to pass a `Cable` then a `ConnectionInfo` object later in the call stack, and I can redo some of these changes. 

I'm not in love with declaring the `ConnectionInfo` here and then assigning the `connectionInfoPtr` member variable to it. For a situation like this I feel like it would make sense to have a `ConnectionInfo` member variable that gets updated as the cable is created and becomes connected. I guess if a `ConnectionInfo` member variable was implemented it could replace the current `startProc`, `endProc`..  member variables.

I thought I would ask what you thought before I went and modified any more but let me know if you have any thoughts on how this should look.

Reply:
Yeah, passing a pointer throughout is totally fine as well.

There's a couple of interesting points here...
- I'm not sure it makes sense for the `Cable` to own the `ConnectionInfo` as a `std::unique_ptr>`. If `ConnectionInfo` were a big class, it would probably make sense to have this method return a `std::unique_ptr<ConnectionInfo>` and have everything downstream own the pointer. _However_, since `ConnectionInfo` is just a tiny little struct, I think it's fine to just have this method return the struct. Typically for small structs like this the compiler can optimize out any copy operations that would otherwise need to happen when returning a class/struct from a function.
- That said, I think your point about having a `ConnectionInfo` member variable makes even more sense, since at the moment, the `Cable` has a bunch of member variables that basically duplicate the role of `ConnectionInfo` anyway, so probably the way to go is something like:
```cpp
class Cable
{
    // gets updated whenever the cable's connection changes,
    // and replaces current member variables like startProc, endProc
    ConnectionInfo connectionInfo;

    // maybe we don't need this if the member variable is public?
    ConnectionInfo getConnectionInfo() const noexcept
    {
        return connectionInfo;
    }
};
```

Reply:
Oh also, even though it probably won't matter, since BYOD is using C++20, we can do [designated initialization](https://www.cppstories.com/2021/designated-init-cpp20/), so we can do this like:
```cpp
ConnectionInfo connection
{
    .startProc = startProc,
    .endProc = endProc,
    .startPort - startIdx,
    .endPort = endIdx,
};
```